core/types/transaction_marshalling.go

```diff
@@ -122,6 +122,20 @@ func (t *Transaction) MarshalJSON() ([]byte, error) {
 		enc.V = (*hexutil.Big)(tx.V)
 		enc.R = (*hexutil.Big)(tx.R)
 		enc.S = (*hexutil.Big)(tx.S)
+	case *CeloDynamicFeeTxV2:
+		enc.ChainID = (*hexutil.Big)(tx.ChainID)
+		enc.AccessList = &tx.AccessList
+		enc.Nonce = (*hexutil.Uint64)(&tx.Nonce)
+		enc.Gas = (*hexutil.Uint64)(&tx.Gas)
+		enc.MaxFeePerGas = (*hexutil.Big)(tx.GasFeeCap)
+		enc.MaxPriorityFeePerGas = (*hexutil.Big)(tx.GasTipCap)
+		enc.FeeCurrency = tx.FeeCurrency // todo: check if needs deep copy
```

Comment:
This didn't bring any issue that we are aware of, and is not something we are modifying or anything like it. I don't know the actual purpose of that comment, but I could copy that just in case.
The `To` field does a deep copy, so probably should be the correct scenario (I just copy pasted the other instances)